src/prefect/server/api/workers.py

```diff
@@ -335,6 +335,12 @@ async def get_scheduled_flow_runs(
             limit=limit,
         )
 
+        await models.workers._update_deployment_last_polled_from_work_pool_name(
+            session=session,
+            work_pool_name=work_pool_name,
+            db=db,
+        )
```

Comment:
This may not be necessary. We record work queue polls in this route already, and we should be able to use the work queue last polled time to inform deployment status.

Reply:
You're right, it wasn't. I've just added tests instead :)